projects/backend/controllers/weather.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response as ExpressResponse } from 'express'
+import fetch, { Response } from 'node-fetch'
+
+const accuWeatherBaseUrl = 'https://weather.guardianapis.com'
+
+const accuWeatherApiKey: string | undefined =
+    process.env.ACCUWEATHER_API_KEY || undefined
+
+const getLocationKey = (ipAddress: string): Promise<Response> => {
+    return fetch(
+        `${accuWeatherBaseUrl}/locations/v1/cities/ipaddress?apikey=${accuWeatherApiKey}&details=true&metric=true&language=en-gb&q=${ipAddress}`,
+    )
+}
+
+const getWeatherForecastForLocation = (
+    locationKey: string,
+): Promise<Response> => {
+    return fetch(
+        `${accuWeatherBaseUrl}/forecasts/v1/hourly/12hour/${locationKey}.json?apikey=${accuWeatherApiKey}&details=true&metric=true&language=en-gb`,
+    )
+}
+
+export const weatherController = (req: Request, res: ExpressResponse) => {
+    const ipAddress: string = req.params.ipAddress
```

Comment:
I think cloudfront will break this, if we want to show ip geolocated (instead of device) we will need to parse the X-Forwarded-For header. For now we might be better off hardcoding this to London.